File: java/src/museo/Escultura.java
package museo;

import java.time.LocalDate;

public class Escultura extends ObraArte {

    private final String material;
    private final String estilo;

    public Escultura(
            int idObra,
            String titulo,
            String autor,
            String periodo,
            double valorEconomico,
            LocalDate fechaCreacion,
            LocalDate fechaIngreso,
            String estado,
            String material,
            String estilo
    ) {
        super(idObra, titulo, autor, periodo, valorEconomico, fechaCreacion, fechaIngreso, estado);
        this.material = material;
        this.estilo = estilo;
    }

    @Override
    public String mostrarDetalle() {
        return String.format(
                "Escultura: %s (%s, %s) - material: %s, estilo: %s",
                getTitulo(), getAutor(), getPeriodo(), material, estilo
        );
    }
}

File: javascript/src/museo/escultura.js
import { ObraArte } from "./obraArte.js";

export class Escultura extends ObraArte {
  constructor(
    idObra,
    titulo,
    autor,
    periodo,
    valorEconomico,
    fechaCreacion,
    fechaIngreso,
    estado,
    material,
    estilo
  ) {
    super(idObra, titulo, autor, periodo, valorEconomico, fechaCreacion, fechaIngreso, estado);
    this.material = material;
    this.estilo = estilo;
  }

  mostrarDetalle() {
    return `Escultura: ${this.getTitulo()} (${this.getAutor()}, ${this.getPeriodo()}) - material: ${this.material}, estilo: ${this.estilo}`;
  }
}
